psydac/fem/splines.py

```diff
@@ -134,7 +135,9 @@ def __init__( self, degree, knots=None, grid=None, multiplicity=None, parent_mul
         self._histopolation_grid   = unroll_edges(self.domain, self.ext_greville)
 
         # Create space of spline coefficients
-        self._vector_space = StencilVectorSpace([nbasis], [self._pads], [periodic])
+        domain_decomposition = DomainDecomposition([self._ncells], [periodic])
+        cart     = CartDecomposition(domain_decomposition, [nbasis], [np.array([0])],[np.array([nbasis-1])], [self._pads], [multiplicity])
+        self._vector_space = StencilVectorSpace( cart )
```

Comment:
Does every 1D spline space has its own 1D `CartDecomposition` now?

Reply:
yes, because each 1D spline has its own `vector_space` and now we can create the `vector_space` only by passing it a `cart`

Reply:
How does this work in practice when the code runs in parallel?

If we have a `TensorFemSpace` defined over a 2D domain, this will have a 2D `DomainDecomposition` and a 2D `CartDecomposition`... But what about the 1D spline spaces this is constructed from? Will these have a 1D `DomainDecomposition` and a 1D `CartDecomposition` compatible with the 2D parallel objects above? Or will they have 1D objects that are serial?

Reply:
The  1D  SplineSpace objects are always serial, we can make them parallel, but this needs more work

Reply:
I am fine with them being serial! I just wonder why we need to create many 1D objects of type `DomainDecomposition` and `CartDecomposition`, if they are not used in parallel...

Reply:
Unless we want to remove the vector_space attribute, we only have one way to do it

Reply:
Good point! I have been thinking about the fact that we should separate the global FEM data from the local-to-process information. Right now `self._vector_space` contains both, but it should probably only contain the local-to-process information... the result is that we would not need this attribute for 1D spline spaces that are not distributed.

We should keep this in mind for the future!